### ISSUE TITLE
FreeRDP proxy: freerdp_post_connect failed

### ISSUE THREAD

**Issue by maordadush** [FreeRDP/FreeRDP#8040]
To Reproduce
connect to the target with via the FreeRDP proxy server.

Application details
FreeRDP proxy version: 3.0.0-dev, Git commit: 3.0.0 (master)
OS version connecting to (server side): Windows 10

Environment:
OS: Windows 10
Architecture: amd64

proxy config:

```
[Security]
ServerTlsSecurity = TRUE
ServerRdpSecurity = FALSE
ServerNlaSecurity = FALSE
ClientTlsSecurity = TRUE
ClientRdpSecurity = FALSE
ClientNlaSecurity = TRUE
ClientAllowFallbackToTls = TRUE

[Channels]
GFX = TRUE
DisplayControl = TRUE
Clipboard = TRUE
AudioOutput = TRUE
RemoteApp = TRUE
DeviceRedirection = TRUE
VideoRedirection = TRUE
CameraRedirection = TRUE
PassthroughIsBlacklist = TRUE
; a list of comma seperated static channels that will be proxied. This feature is useful,
; for example when there's a custom static channel that isn't implemented in freerdp/proxy, and is needed to be proxied when connecting through the proxy.
Passthrough =

[Clipboard]
TextOnly = FALSE
MaxTextLength = 10 # 0 for no limit.

[GFXSettings]
DecodeGFX = TRUE
```

I got the following error:

```
[WARN][com.freerdp.common] - FreeRDP_ChannelDefArray::len expected to be >= 31, but have 6
[WARN][com.freerdp.core.update] - Tried to resize the session, but RNS_UD_CS_SUPPORT_MONITOR_LAYOUT_PDU is not supported by client
[ERROR][com.freerdp.core] - freerdp_abort_connect_context:freerdp_set_last_error_ex ERRCONNECT_CONNECT_CANCELLED [0x0002000B]
[ERROR][com.freerdp.core] - freerdp_post_connect failed
[ERROR][com.freerdp.core] - freerdp_connect:freerdp_set_last_error_ex ERRCONNECT_POST_CONNECT_FAILED [0x00020003]
[DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_ACTIVE] reset finalize_sc_pdus
[DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_ACTIVE --> CONNECTION_STATE_INITIAL
[ERROR][proxy.client] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_client_connect]: failed to connect with NLA. retrying to connect without NLA
[DEBUG][com.freerdp.core.rdp] - [CONNECTION_STATE_INITIAL] reset finalize_sc_pdus
[DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_INITIAL --> CONNECTION_STATE_INITIAL
[DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling TLS security: TRUE
[DEBUG][com.freerdp.core.nego] - Enabling NLA security: FALSE
[DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_TLS
[DEBUG][com.freerdp.core.nego] - Attempting TLS security
[DEBUG][com.freerdp.core] - freerdp_tcp_is_hostname_resolvable:freerdp_set_last_error_ex resetting error state
[DEBUG][com.freerdp.core] - freerdp_tcp_default_connect:freerdp_set_last_error_ex resetting error state
[DEBUG][com.freerdp.core] - connecting to peer xxx.xxx.xxx.xxx
[DEBUG][com.freerdp.core.nego] - RequestedProtocols: 1
[DEBUG][com.freerdp.core.nego] - RDP_NEG_RSP::flags = { [0x1f] |EXTENDED_CLIENT_DATA_SUPPORTED|DYNVC_GFX_PROTOCOL_SUPPORTED|RDP_NEGRSP_RESERVED|RESTRICTED_ADMIN_MODE_SUPPORTED|REDIRECTED_AUTHENTICATION_MODE_SUPPORTED }
[DEBUG][com.freerdp.core.nego] - selected_protocol: 1
[DEBUG][com.freerdp.core.nego] - Negotiated TLS security
[DEBUG][com.freerdp.core.nego] - nego_security_connect with PROTOCOL_SSL
[DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_NEGO --> CONNECTION_STATE_MCS_CONNECT
[DEBUG][com.freerdp.core.gcc] - [gcc_write_client_monitor_data] MonitorCount=0
[DEBUG][com.freerdp.core.gcc] - [gcc_write_client_monitor_data] FINISHED
[DEBUG][com.freerdp.core.gcc] - Server rdp encryption method: NONE
[DEBUG][com.freerdp.core.connection] - rdp_client_transition_to_state CONNECTION_STATE_MCS_CONNECT --> CONNECTION_STATE_MCS_ATTACH_USER
[DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[INFO][proxy.server] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_server_handle_peer]: starting shutdown of connection
[INFO][proxy.server] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_server_handle_peer]: stopping proxy's client
[ERROR][com.freerdp.core] - freerdp_abort_connect_context:freerdp_set_last_error_ex ERRCONNECT_CONNECT_CANCELLED [0x0002000B]
[INFO][proxy.server] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_server_handle_peer]: freeing server's channels
[INFO][proxy.server] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_server_handle_peer]: freeing proxy data
[DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[ERROR][proxy.client] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_client_connect]: pf_client_connect_without_nla failed!
[DEBUG][proxy.client] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_client_client_stop]: aborting client connection
[DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[DEBUG][proxy.server] - [SessionID=BCA188D1381266D9F9E0E291A58AFF64][pf_server_handle_peer]: Removed peer, 0 connected

```

When I'm connecting directly to the remote target with the same username and password is working.

**Comment by akallabeth** [comment]
Can we have some client details please?
`[WARN][com.freerdp.core.update] - Tried to resize the session, but RNS_UD_CS_SUPPORT_MONITOR_LAYOUT_PDU is not supported by client` looks like there is some negotiation not working well

**Comment by maordadush** [comment]
![image](https://user-images.githubusercontent.com/8192075/177305175-218c9485-6ebc-46dd-8d33-40596c58d4b9.png)
windows 10 21H1 19043.1766

**Comment by maordadush** [comment]
yes, with bitmap updates

**Comment by maordadush** [comment]
also I noticed that device redirection not working

**Comment by akallabeth** [comment]
@maordadush Fixed with #8046  thanks for raising the issue.
